### ISSUE TITLE
How to monitor training process using a development set?

### ISSUE THREAD

**Issue by pengding1994** [dmlc/xgboost#3310]
For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

## Environment info
Operating System: Windows 7

Compiler: Visual Studio 2013

Package used (python/R/jvm/C++): C++

`xgboost` version used: 0.71

If installing from source, please provide 

1. The commit hash (`git rev-parse HEAD`)
2. Logs will be helpful (If logs are large, please upload as attachment).

If you are using jvm package, please 

1. add [jvm-packages] in the title to make it quickly be identified
2. the gcc version and distribution

If you are using python package, please provide

1. The python version and distribution
2. The command to install `xgboost` if you are not installing from source

If you are using R package, please provide

1. The R `sessionInfo()`
2. The command to install `xgboost` if you are not installing from source

## Steps to reproduce

1.
2.
3.

## What have you tried?

I am using the cli_main.cc in Visual Studio 2013 for a multiclass problem. I tried to use a set of validation data to monitor the training process and I noticed there is a parameter for setting path of development set. my config file is shown below. However, no matter how do I change eval[train], results make no difference. Do I miss any important parameters to enable the monitoring process or do I misunderstand the process? Thank you for any help!

# General Parameters, see comment for each definition
# can be gbtree or gblinear
booster = gbtree 
# choose logistic regression loss function for binary classification
objective = multi:softprob

# Tree Booster Parameters
# step size shrinkage
eta = 0.15
# minimum loss reduction required to make a further partition
gamma = 1.5
# minimum sum of instance weight(hessian) needed in a child
min_child_weight = 1 
# maximum depth of a tree
max_depth = 12
max_delta_step = 10

# Task Parameters
# the number of round to do boosting
num_round = 40
# 0 means do not save any model except the final round model
save_period = 0 
# The path of training data
data = D:\xgboost\demo\data\rrk_tf\rrk_tf_train.txt 
# The path of validation data, used to monitor training process, here [test] sets name of the validation set
eval[test] =D:\xgboost\demo\data\rrk_tf\rrk_tf_supervisor.txt  
# The path of test data 
test:data = D:\xgboost\demo\data\rrk_tf\rrk_tf_test.txt     
nthread = 4
task = train
eval_metric = merror
eval_train = 1
#model_in = 0040.model
name_dump = dump.nice.txt
silent = 0
